web/frontend/src/api/navigation.ts

```diff
@@ -1,211 +1,79 @@
+            radius: (geometry.sphere.radiusMm ?? 0) / 1000,
             pose,
           } satisfies SphereGeometry;
 
-        } else if (geometry.hasCapsule()) {
-          const capsule = geometry.getCapsule();
+        } else if (geometry.capsule) {
+          const { capsule } = geometry;
 
           return {
             type: 'capsule',
-            radius: (capsule?.getRadiusMm() ?? 0) / 1000,
-            length: (capsule?.getLengthMm() ?? 0) / 1000,
+            radius: (capsule?.radiusMm ?? 0) / 1000,
+            length: (capsule?.lengthMm ?? 0) / 1000,
             pose,
           } satisfies CapsuleGeometry;
 
         }
 
-        notify.danger('An unsupported geometry was encountered in an obstacle', JSON.stringify(geometry.toObject()));
+        notify.danger('An unsupported geometry was encountered in an obstacle', JSON.stringify(geometry));
         throw new Error(
-          `An unsupported geometry was encountered in an obstacle: ${JSON.stringify(geometry.toObject())}`
+          `An unsupported geometry was encountered in an obstacle: ${JSON.stringify(geometry)}`
         );
       }),
     } satisfies Obstacle;
   });
 };
-
-export const getWaypoints = async (robotClient: Client, name: string): Promise<Waypoint[]> => {
-  const req = new navigationApi.GetWaypointsRequest();
-  req.setName(name);
-
-  rcLogConditionally(req);
-
-  const response = await new Promise<{ getWaypointsList(): navigationApi.Waypoint[] } | null>((resolve, reject) => {
-    robotClient.navigationService.getWaypoints(req, (error, res) => {
-      if (error) {
-        reject(error);
-      } else {
-        resolve(res);
-      }
-    });
-  });
-
-  return formatWaypoints(response?.getWaypointsList() ?? []);
-};
-
-export const removeWaypoint = async (robotClient: Client, name: string, id: string) => {
-  const request = new navigationApi.RemoveWaypointRequest();
-  request.setName(name);
-  request.setId(id);
-
-  rcLogConditionally(request);
-
-  const response = await new Promise<navigationApi.RemoveWaypointResponse | null>((resolve, reject) => {
-    robotClient.navigationService.removeWaypoint(request, (error, res) => {
-      if (error) {
-        reject(error);
-      } else {
-        resolve(res);
-      }
-    });
-  });
-
-  return response?.toObject();
-};
-
-export const getLocation = async (robotClient: Client, name: string) => {
-  const request = new navigationApi.GetLocationRequest();
-  request.setName(name);
-
-  rcLogConditionally(request);
-
-  const response = await new Promise<navigationApi.GetLocationResponse | null>((resolve, reject) => {
-    robotClient.navigationService.getLocation(request, (error, res) => {
-      if (error) {
-        reject(error);
-      } else {
-        resolve(res);
-      }
-    });
-  });
-
-  const location = response?.getLocation();
-  const lat = location?.getLatitude();
-  const lng = location?.getLongitude();
-
-  rcLogConditionally(response?.getLocation()?.toObject());
```

Comment:
These logs is needed because of some strange responses that we're getting from the server. Could you re-add them where we call this function?